esphome/components/hdmi_cec/hdmi_cec.cpp

```diff
@@ -0,0 +1,155 @@
+#ifdef USE_ARDUINO
+
+#include "hdmi_cec.h"
+#include "esphome/core/log.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace hdmi_cec {
+
+static const char *const TAG = "hdmi_cec";
+
+void message_to_debug_string(char *message, const unsigned char *buffer, int count) {
+  for (int i = 0; i < count; i++) {
+    sprintf(&(message[i * 3]), "%02X", buffer[i]);
+    if (i < count - 1) {
+      sprintf(&(message[i * 3 + 2]), ":");
+    }
+  }
+}
+
+bool HdmiCec::LineState() {
+  int state = this->pin_->digital_read();
+  return state != LOW;
```

Comment:
```suggestion
  return !this->pin_->digital_read();
```

`digital_read()` (in contrast to `digitalRead` from arduino) returns a bool